slick/src/main/scala/slick/jdbc/JdbcActionComponent.scala

```diff
@@ -208,17 +203,27 @@ trait JdbcActionComponent extends SqlActionComponent { self: JdbcProfile =>
         case ParameterSwitch(cases, default) =>
           findSql(cases.find { case (f, n) => f(param) }.map(_._2).getOrElse(default))
       }
-      (tree match {
+      val (comment, tree2) = tree match {
+        case Comment(comment, child) =>
+          if(comment.isEmpty) {
+            ("", child)
+          } else {
+            (s"/* $comment */\n", child)
+          }
+        case _ =>
+          ("", tree)
```

Comment:
You can move the emptiness check to a "pattern guard" to fall through to the default case if the comment is empty. Also probably a good idea to handle whitespace too:

```suggestion
        case Comment(comment, child) if comment.trim.nonEmpty =>
          (s"/* $comment */\n", child)
        case _ =>
          ("", tree)
```